plugins/exporter.koplugin/target/my_clippings.lua

```diff
@@ -11,25 +11,41 @@ local ClippingsExporter = require("base"):new {
 }
 
 local function format(booknotes)
-    local content = ""
+    local tbl = {}
+
     for ___, entry in ipairs(booknotes) do
         for ____, clipping in ipairs(entry) do
             if booknotes.title and clipping.text then
-                content = content .. booknotes.title .. "\n"
-                local header = T(_("- Your highlight on page %1 | Added on %2"), clipping.page, os.date("%A, %B %d, %Y %I:%M:%S %p", clipping.time)) .. "\n\n"
-                content = content .. header
-                content = content .. clipping.text
-                content = content .. "\n==========\n"
+                table.insert(tbl, booknotes.title)
+
+                local header = T(_("- Your highlight on page %1 | Added on %2"), clipping.page,
+                    os.date("%A, %B %d, %Y %I:%M:%S %p", clipping.time))
+                table.insert(tbl, header)
+
+                table.insert(tbl, "")
+                table.insert(tbl, clipping.text)
+                table.insert(tbl, "==========")
+
                 if clipping.note then
-                    content = content .. booknotes.title .. "\n"
-                    header = T(_("- Your note on page %1 | Added on %2"), clipping.page, os.date("%A, %B %d, %Y %I:%M:%S %p", clipping.time)) .. "\n\n"
-                    content = content .. header .. clipping.note
-                    content = content .. "\n==========\n"
+                    table.insert(tbl, booknotes.title)
+
+                    header = T(_("- Your note on page %1 | Added on %2"), clipping.page,
+                        os.date("%A, %B %d, %Y %I:%M:%S %p", clipping.time))
+                    table.insert(tbl, header)
+
+                    table.insert(tbl, "")
+                    table.insert(tbl, clipping.note)
+                    table.insert(tbl, "==========")
                 end
             end
         end
     end
-    return content
+
+    -- Add an extra newline to keep a consistent format when exporting
+    -- multiple books' data; if this is missing, the title is written
+    -- after the previous book's last "=========" as "==========New book title"
```

Comment:
I think it's fairly self-explanatory, but if we feel a comment is in order I'd just keep it brief. Something like:

```suggestion
    -- Ensure a newline after the previous book's last "=========".
```